### ISSUE TITLE
Placing a container next to an item pipe will not process servos

### ISSUE THREAD

**Issue by Qendolin** [Technici4n/Modern-Dynamics#9]
Depending on how often and when the configuration of a pipe has been changed, placing a container next to it will not make servos send items to it.

The details to when items are transferred and when not are not quite clear to me, so here are some examples:

## No Transfer:
![image](https://user-images.githubusercontent.com/32160662/168496522-577563e7-3200-429c-842f-aca8674a144d.png)
Build Order: 
1. Lapis Chest (with items)
2. Pipe
3. Servo
4. Diamond Chest
Works every time

![image](https://user-images.githubusercontent.com/32160662/168496653-c9c7a260-c96a-41b9-b76e-c830a6e7219e.png)
Build Order: 
1. Lapis Chest (with items)
2. Pipes (any order)
3. Servo
4. Diamond Chest
Works every time

## Transfer:

![image](https://user-images.githubusercontent.com/32160662/168496860-d2fc8d46-191c-45bf-95c6-89dad41e4c9e.png)
Build Order: 
1. Lapis Chest (with items)
2. Glas pipe
3. Gold pipe
4. Servo
5. Break gold pipe
6. Diamond Chest
Works every time

## Toggle Transfer:

![image](https://user-images.githubusercontent.com/32160662/168496940-fd5133c3-ae2e-4e30-8949-200ebbc2b6cd.png)
Build Order: 
1. Lapis Chest (with items)
2. Pipe
4. Servo
5. Diamond Chest
6. No transfer
7. Use the wrench twice on the pipe (disconnect and connect)
8. Wait for transfer
9. Break the diamond chest and place it again
10. Still transfers
11. Use the wrench twice on the pipe (disconnect and connect)
12. Break the diamond chest immoderately before any items enter the pipe
13. Place the diamond chest
14. No transfer
Works every time

Conclusion: Toggling the pipe connection to the chest will fix the transfer for future placements. **Unless** the destination chest is broken before any items enter the pipe in wich case the transfer will again be broken for future placements.

![image](https://user-images.githubusercontent.com/32160662/168497283-ce87fa72-6912-4c4b-b34b-aa6fc363d60a.png)
Build Order: 
1. Lapis Chest (with items)
2. Pipe
4. Servo
5. Inhibitor
6. Diamond Chest
7. No Transfer
Works sometimes. (Depending on what, I don't know)

**Comment by Technici4n** [comment]
Thanks a lot for your testing, it looks like the issue I fixed a few minutes ago. :smile:

It was caused by item pipes forgetting to invalidate the cached paths when the connections of a pipe to adjacent inventories changed. Should be fixed with the latest commit (a build should be available at https://github.com/Technici4n/Modern-Dynamics/actions/runs/2328763593, I'll do a proper release tomorrow).
